tests/bsim/bluetooth/mesh/src/test_dfu.c

```diff
@@ -435,25 +435,52 @@ static void target_prov_and_conf_default(void)
 	target_prov_and_conf(addr, bind_params, ARRAY_SIZE(bind_params));
 }
 
+static struct bt_mesh_dfu_slot *slot_reserve_and_set(size_t size, uint8_t *fwid, size_t fwid_len,
+						     uint8_t *metadata, size_t metadata_len)
+{
+	struct bt_mesh_dfu_slot *new_slot = bt_mesh_dfu_slot_reserve();
+
+	if (!new_slot) {
+		return NULL;
+	}
+
+	int err = bt_mesh_dfu_slot_fwid_set(new_slot, fwid, fwid_len);
+
+	if (err) {
+		(void)bt_mesh_dfu_slot_release(new_slot);
```

Comment:
This I think shouldn't be needed as the test should fail if `bt_mesh_dfu_slot_fwid_set` (or `bt_mesh_dfu_slot_info_set`, or `bt_mesh_dfu_slot_commit` below) fails.